src/components/AddRemoteDialogue.tsx

```diff
@@ -0,0 +1,198 @@
+  /**
+   * New remote name and url pair
+   */
+  newRemote: Git.IGitRemote;
+  /**
+   * List of known remotes
+   */
+  existingRemotes: Git.IGitRemote[] | null;
+}
+
+export class AddRemoteDialogue extends React.Component<
+  IAddRemoteDialogueProps,
+  IAddRemoteDialogueState
+> {
+  constructor(props: IAddRemoteDialogueProps) {
+    super(props);
+    this.state = {
+      newRemote: {
+        name: '',
+        url: ''
+      },
+      existingRemotes: null
+    };
+  }
+
+  async componentDidMount(): Promise<void> {
+    const remotes = await this.props.model.getRemotes();
+    this.setState({ existingRemotes: remotes });
+  }
+
+  render(): JSX.Element {
+    return (
+      <Dialog
+        classes={{
+          paper: remoteDialogClass
+        }}
+        open={true}
+        onClose={this.props.onClose}
+      >
+        <div className={titleWrapperClass}>
+          <p className={titleClass}>{this.props.trans.__('Manage Remotes')}</p>
+          <button className={closeButtonClass}>
+            <ClearIcon
+              titleAccess={this.props.trans.__('Close this dialog')}
+              fontSize="small"
+              onClick={() => {
+                this.props.onClose();
+              }}
+            />
+          </button>
+        </div>
+        <div className={contentWrapperClass}>
+          <label className={remoteDialogInputClass}>
+            <span>
+              {this.props.trans.__(
+                'Enter a new remote repository name and url'
+              )}
+            </span>
+            <input
+              type="text"
+              placeholder={this.props.trans.__('name')}
+              onChange={event =>
+                this.setState({
+                  newRemote: {
+                    ...this.state.newRemote,
+                    name: event.target.value
+                  }
+                })
+              }
+            />
+            <input
+              type="text"
+              placeholder={this.props.trans.__('Remote Git repository URL')}
+              onChange={event =>
+                this.setState({
+                  newRemote: {
+                    ...this.state.newRemote,
+                    url: event.target.value
+                  }
+                })
+              }
+            />
+          </label>
+
+          {this.props.warningContent && (
+            <div className="jp-RemoteDialog-warning">
+              {this.props.warningContent}
+            </div>
+          )}
+
+          <DialogActions className={actionsWrapperClass}>
+            <input
+              className={classes(buttonClass, cancelButtonClass)}
+              type="button"
+              title={this.props.trans.__(
+                'Close this dialog without adding a remote repo'
+              )}
+              value={this.props.trans.__('Cancel')}
+              onClick={() => {
+                this.props.onClose();
+              }}
+            />
```

Comment:
Let's remove the cancel button as it feels awkward.

```suggestion
```

Reply:
Removed the cancel button